Populate RX frequency

njgraham opened the pull request.

Commit:
Extract the frequency code from the pcori_basecode column.

 -

Commit:
Populate pcornet_med terms with local RX frequency modifier leaves

 - Remove erroneous comment.

Commit:
Replace reference to local schema/table with substitution variables.

dckc commented on Oracle/PCORNetLoader_ora.sql:
```diff
@@ -1471,7 +1471,7 @@ insert into prescribing (
 )
 select distinct  m.patient_num, m.Encounter_Num,m.provider_id,  m.start_date order_date,  to_char(m.start_date,'HH:MI'), m.start_date start_date, m.end_date, mo.pcori_cui
-    ,quantity.nval_num quantity, refills.nval_num refills, supply.nval_num supply, freq.pcori_basecode frequency, 
+    ,quantity.nval_num quantity, refills.nval_num refills, supply.nval_num supply, substr(freq.pcori_basecode, instr(freq.pcori_basecode, ':') + 1, 2) frequency, 
```
why the substr() here? Are we putting something other than exactly what goes in the CDM frequency column? Why?

njgraham replied:
@dckc, it's because the SCILHS medication ontology has things like `RX_FREQUENCY:02` in the pcori_basecode column.  See [pcornet_med.zip](https://github.com/SCILHS/scilhs-ontology/blob/master/Ontology/pcornet_med.zip)

dckc replied:
Then how did the original code work for anybody else? Are you suggesting that it did not?

njgraham replied:
As far as I can tell it either didn't work or something other than the [scilhs-ontology](https://github.com/SCILHS/scilhs-ontology) was used.

dckc replied:
I raised https://github.com/SCILHS/i2p-transform/issues/25 about this.

Commit:
Added PATID and ENCOUNTERID indexes to tables that contain these fields.

Commit:
Added DROP INDEX statements to prevent errors if indexes already exist.

Commit:
Merge branch 'mprittie_master' into rx_frequency

Commit:
Use PMN_DROPSQL procedure to drop indexes to avoid throwing errors if the index doesn't exist.

Commit:
Merge branch 'mprittie_master' into rx_frequency

Commit:
Merge pull request #2 from njgraham/mprittie_master

Use PMN_DROPSQL procedure to drop indexes...

Commit:
Copy medication facts into their own table - reference while building prescribing.

 - Putting medication facts in their own table will hopefully reduce needed system resources when building  the prescribing table.

Commit:
Add early test to make sure the medication fact table is populated.

Commit:
Merge remote-tracking branch 'mprittie/master' into rx_frequency

Commit:
Drop indexes at the beginning of each procedure rather than the end.

 - Inserting with existing indexes leads to poor performance in practice.
 - Thanks to Mike for review.

Commit:
Add more indexes and gather table statistics to hopefully improve performance.

 - Added missing schema for insert into pcornet_med.

Commit:
Procedure to gather table statistics to consolidate parameters.

Commit:
Break apart admit source query to improve performance.

 - Add indexes/gather table statistics to help Oracle pick better plans.

Commit:
Use the fact table primary key to join prescribing table with basis, freq, etc.

 - Thanks to Russ for pointing out that we may have many medication facts per encounter/concept code.

njgraham commented:
See [GPC ticket 496](https://informatics.gpcnetwork.org/trac/Project/ticket/496#comment:12) for details.  Note that several performance changes got included on the rx_frequency branch - I hope that's ok.

dckc commented:
This code is the basis of our production build 80 of Aug 11. Let's deal with any changes to it in later issues.
